### DISCUSSION TITLE
how i can use withSum that should be eager loaded on every query like withCount

### DISCUSSION THREAD

**Discussion by samehdoush** [laravel/framework#44515] [General] [answered]
ex : User Model 
```
 protected $withCount = [];
 protected $withSum = [];
```
I need it on every query 
Are there any ideas or help for this?

**Comment by kverstae** [comment, marked as answer]
A possible solution would be to add a global query scope (https://laravel.com/docs/9.x/eloquent#global-scopes) to your user model (or whatever model you want).
This would look something like this:

```php
// CountAndSumScope

class CountAndSumScope implements Scope
{
    public function apply(Builder $builder, Model $model)
    {
        $builder->withCount([...])->withSum([...]);
    }
}

// User

protected static function booted()
{
    static::addGlobalScope(new CountAndSumScope);
}
```

This code is untested, but I think this idea could work
